cds/modules/migrator/records.py

```diff
@@ -125,3 +131,58 @@ def prepare_revisions(self):
             self.revisions.append(self._prepare_intermediate_revision(i))
 
         self.revisions.append(self._prepare_final_revision(it[-1]))
+
+
+class CDSRecordDumpLoader(RecordDumpLoader):
+    """Migrate a CDS record."""
+
+    @classmethod
+    def create_files(cls, record, files, existing_files):
+        """Create files.
+
+        This method is currently limited to a single bucket per record.
+        """
+        default_bucket = None
+        # Look for bucket id in existing files.
+        for f in existing_files:
+            if 'bucket' in f:
+                default_bucket = f['bucket']
+                break
+
+        # Create a bucket in default location if none is found.
+        if default_bucket is None:
+            b = Bucket.create()
+            BucketTag.create(b, 'record', str(record.id))
+            default_bucket = str(b.id)
+            db.session.commit()
+        else:
+            b = Bucket.get(default_bucket)
+
+        record['_files'] = []
+        for key, meta in files.items():
+            obj = cls.create_file(b, key, meta)
+            ext = splitext(obj.key)[1].lower()
+            if ext.startswith('.'):
+                ext = ext[1:]
+            file_info = dict(
+                bucket=str(obj.bucket.id),
+                key=obj.key,
+                version_id=str(obj.version_id),
+                size=obj.file.size,
+                checksum=obj.file.checksum,
+                type=ext,
+            )
+            last_ver = meta[-1]
+            if 'recid' in last_ver and 'recids_doctype' in last_ver:
+                rec_docs = [rec_doc[1] for rec_doc in last_ver['recids_doctype']
+                            if rec_doc[0] == last_ver['recid']]
+                if rec_docs:
+                    file_info['doctype'] = rec_docs[0]
```

Comment:
maybe better something like 
```
file_info['doctype'] = rec_docs[0] if rec_docs else ''
```
otherwise we will end up with files with different keys and it makes UI handling a bit more complex